File: fixtures/scale/ruby/large/lib/check_pairswap.rb
require_relative 'pairswap_encode'
require_relative 'pairswap_decode'
require_relative 'corpus_codec'

def check_pairswap
  corpus_codec.each do |text|
    packed = pairswap_encode(text)
    return false if pairswap_decode(packed) != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_pairswap.py
from cipherlab.ciphers.pairswap import pairswap_encode, pairswap_decode
from cipherlab.specs.pairswap_spec import pairswap_spec
from cipherlab.corpus.corpus_codec import corpus_codec


def check_pairswap():
    spec = pairswap_spec()
    for text in corpus_codec():
        packed = pairswap_encode(text)
        if pairswap_decode(packed) != text:
            return False
    return spec["category"] == "codec"
